sources/core/Stride.Core.DiagnosticsServices/PropertyAttributeFinder.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using System.Reflection;
+using Microsoft.CodeAnalysis.CSharp;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace StrideDiagnostics;
+internal class PropertyAttributeFinder
+{
+    /// <summary>
+    /// Walks through a base class of a class and retrieves all allowed Properties.
+    /// Then it tries to get it's own base class and get from it all allowed Properties recursively.
+    /// All the Properties get summed up.
+    /// If they fulfill :
+    ///     Proper Accessors <see cref="PropertyHasAllowedAccessors(IPropertySymbol)"/>
+    ///     Shouldnt be Ignored <see cref="HasDataMemberIgnoreAttribute(IPropertySymbol)"/>
+    /// </summary>
+    /// <param name="currentBaseType">The base class which the ClassDeclarationSyntax has</param>
+    /// <returns>All allowed Properties in any base class in the inheritance tree</returns>
+    public static IEnumerable<IPropertySymbol> FilterBasePropertiesRecursive(ref INamedTypeSymbol currentBaseType)
+    {
+        List<IPropertySymbol> result = new List<IPropertySymbol>();
+        while (currentBaseType != null)
+        {
+            result.AddRange(currentBaseType.GetMembers().OfType<IPropertySymbol>().Where(PropertyHasAllowedAccessors));
+            currentBaseType = currentBaseType.BaseType;
+        }
+        return result;
+    }
+
+    private static bool PropertyHasAllowedAccessors(IPropertySymbol propertyInfo)
+    {
+        if (propertyInfo == null)
+            return false;
+        return (propertyInfo.SetMethod?.DeclaredAccessibility == Accessibility.Public ||
+                propertyInfo.SetMethod?.DeclaredAccessibility == Accessibility.Internal ||
+                propertyInfo.GetMethod?.ReturnsVoid == true
```

Comment:
When does a getter return void? Did you mean `SetMethod`?

Reply:
i think it returns void if its not existent , i will check it out

Reply:
[f7d15a0](https://github.com/stride3d/stride/pull/1778/commits/f7d15a0e53ee996ea347365f8bd9f13b8b91296a)
it will never be true for normal getters